### PR TITLE
Allow usage of ES6 classes for components

### LINKED ISSUE #78: Support ES6 class components
When using an [ES6 class](https://mithril.js.org/components.html#es6-classes) style component, I get an error like:

```sh
Sample > should have a div:
Class constructor Sample cannot be invoked without 'new'
```

---

Reproduction example:

```js
global.window = require("mithril/test-utils/browserMock.js")()
global.document = window.document

var m = require("mithril") // version 1.1.1
var mq = require('mithril-query') // version 2.2.0
var o = require("mithril/ospec/ospec")

class Sample {
    constructor(vnode) {
	this._text = vnode.attrs.text
    }

    view() {
	return m("div", this._text)
    }
}


o.spec("Sample", function() {
    o("should have a div", function() {
	var output = mq(Sample, {text: "Text"})
	output.should.have(1, "div")
    })
})
```

### PR DESCRIPTION
**Pull request by jmooradi**
Fixes #78

### CODE REVIEW

**Comment by jmooradi** [comment]
Sorry I have been busy, I will try to add test cases and fix the failed checks soon

**Comment by evanwalsh** [comment]
@jmooradi Any progress on this? If not, I can try to push it forward. This is blocking some testing I want to do.

**Comment by jmooradi** [comment]
Should be all good now, sorry for the delay!

File: test.js [hunk 1]
```diff
@@ -440,6 +462,66 @@ describe('components', function () {
...
+      out = mq(new SimpleES6Component(), { name: 'Homer' })
+      out.should.have('div:contains(Hello from Homer)')
+    })
+
+    it('should call onremove on globalonremove', function (done) {
+      ES6Component.prototype.onremove = function () { done() }
+      var out = mq(new ES6Component())
+      out.onremove()
+    })
+  })
```
Comment (StephanHoyer) [inline test.js]: Awesome work! Thanks a lot @jmooradi!

What I miss are some test cases where the same class component is rendered multiple times and the state of it persists. Like
```
var out = mq(new ES6Component())
// do something that changes the state of the component
// check if the component rendered the updated state
out.redraw()
// check if the component still rendered the updated state
```

Or even create a component with a click-handler that mutates it's own state, like a switch or something.

Also what I miss are some tests where a class component is embedded somewhere in the tree, at best als a class component that is embedded in another class component.

But nevertheless, great work, highly appreciated!

Comment (StephanHoyer) [reply test.js]: @jmooradi any progress here?

Comment (StephanHoyer) [reply test.js]: @jmooradi ping

**Pull request merged**